calculator/parser.py

```diff
@@ -139,3 +182,96 @@ def rpn(tokens):
                 ",".join(str(t) for t in tokens)
             )
         )
+
+
+def infix(token_list):
+
+    """
+    Parses Infix notation and returns the equivilant RPN form
+
+    Parameters
+    ==========
+    token_list : list
+        The list of infix tokens
+
+    Returns 
+    --------
+    output : list
+        This is the output queue where the final answer is arranged
+    
+    """
+
+    # Output Queue
+    output = []
+
+    # Operators Stack
+    operators = []
+
+    iteration_num = 1
+
+    # Iterating through each token in the token_list
+    for token in token_list:
+
+        # If the token is the a digit, or:
+        # 1. It has a negative sign in it and the whole number is not just a negative sign
+        # 2. or a decimal point is in the token
+        if (
+            str(token).isdigit()
+            or "-" in str(token)
+            and str(token) != "-"
+            or "." in str(token)
+        ):
+
+            push(token, output)
+            iteration_num += 1
+
+        elif is_function(token):
+            push(token, operators)
+
+        elif token in lexer.OPS:
+            # If there are tokens to be read then
+            # 1. And the operator on teh operator stack has greater precedence (for order of operations)
+            # 2. Or they have equal precedense and the token is left associative
+            # 3. If the operator on the operator stack is not a "("
+
+            while (
+                (peek(operators, "s") != None)
+                and (
+                    (OPS_RANKING[peek(operators, "s")] > OPS_RANKING[token])
+                    or (
+                        OPS_RANKING[peek(operators, "s")] == OPS_RANKING[token]
+                        and OPS_ASSOSIATIVITY[token] == "l"
+                    )
+                )
+                and (peek(operators, "s") != "(")
+            ):
+
+                # Pushing the token on top of the operators stach onto the output queue
+                push(pop(operators, "s"), output)
+                iteration_num += 1
+
+            push(token, operators)
+
+        if token == "(":
+
+            push(token, operators)
+            iteration_num += 1
+
+        elif token == ")":
+
+            # If the top of the operator stack is not a "(" (Left Parenthesis)
+            while peek(operators, "s") != "(":
```

Comment:
These situations where `iteration_num` is incremented inside of `while` loops would imply that the tokens that were delegated are being "caught up" somehow. I would like more explanation in comments when tokens are deferred processing and what conditions would cause them to be caught up.

Reply:
👍